scheduler-driver/src/main/java/gov/nasa/jpl/aerie/scheduler/simulation/SimulationFacade.java

```diff
@@ -146,20 +149,31 @@ public void setActivityTypes(final Collection<ActivityType> activityTypes){
   }
 
   public Map<SchedulingActivityDirectiveId, ActivityDirectiveId> getActivityIdCorrespondence(){
-    return planActDirectiveIdToSimulationActivityDirectiveId;
+    return new HashMap<>(mapSchedulingIdsToActivityIds);
+  }
+
+  public Optional<BidiMap<SchedulingActivityDirectiveId, ActivityDirectiveId>> getBidiActivityIdCorrespondence(){
+    if(initialSimulationResults.isEmpty() || initialPlanHasBeenModified)
+      return Optional.ofNullable(mapSchedulingIdsToActivityIds);
+    else if(initialSimulationResults.isPresent())
+      return initialSimulationResults.get().mapSchedulingIdsToActivityIds();
+    else
+      return Optional.empty();
   }
```

Comment:
why keep both maps ? is `getActivityIdCorrespondence()` still used ?

Reply:
The coversion from bidi to hashmap is done as it is used in SynchronousSchedulerAgent:
```java
    final var schedID_to_simID =
        simulationFacade.getActivityIdCorrespondence();
```

Reply:
I would move the conversion (and thus delete `getActivityIdCorrespondence`) out of this class as the logic you added in getBidiAct... is not present in `getActivityIdCorrespondence`.